### PR TITLE
Fix MatrixDiagV2 gradient for ignored diagonal padding

### LINKED ISSUE #127235: tf.raw_ops.MatrixDiagV2 propagates gradients through ignored diagonal padding
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0, also reproduced on tf 2.22.0-dev20260904

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Mobile device

_No response_

### Python version

Python 3.13.5

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

`tf.raw_ops.MatrixDiagV2` propagates gradients through a diagonal padding value that should be ignored.

The reproducer constructs a diagonal tensor for `k=[0, 1]` and creates a `2 x 2` matrix with `MatrixDiagV2`. Under the legacy left-alignment semantics used by `MatrixDiagV2`, the value `7 * t` is padding for the supplied diagonal band and should not contribute to the output or its derivative.

The forward value is computed correctly as `1.5899999999999999`, matching the scalar expression `t**2 + 3*t - 1`. Therefore, at `t = 0.7`, the expected derivative is `2*t + 3 = 4.4`.

However, reverse-mode autodiff returns `10.0`, which indicates that the ignored padding value is still contributing to the gradient.

Expected behavior?

The gradient of the `tf.raw_ops.MatrixDiagV2` output should ignore diagonal padding values that do not contribute to the forward output. For the reproducer below, the derivative at `t = 0.7` should be close to `4.4`, not `10.0`.

### Standalone code to reproduce the issue

```shell
import os
os.environ["CUDA_VISIBLE_DEVICES"] = ""
os.environ["OMP_NUM_THREADS"] = "1"
os.environ["TF_NUM_INTRAOP_THREADS"] = "1"
os.environ["TF_NUM_INTEROP_THREADS"] = "1"

import tensorflow as tf

def target(t):
    two = tf.constant(2, dtype=t.dtype)
    seven = tf.constant(7, dtype=t.dtype)

    diagonal = tf.stack([
        tf.stack([t * t, seven * t]),
        tf.stack([t, two * t]),
    ])

    y = tf.raw_ops.MatrixDiagV2(
        diagonal=diagonal,
        k=tf.constant([0, 1], dtype=tf.int32),
        num_rows=tf.constant(2, dtype=tf.int32),
        num_cols=tf.constant(2, dtype=tf.int32),
        padding_value=tf.constant(-1, dtype=t.dtype),
    )

    return tf.reduce_sum(y)

x = tf.constant(0.7, dtype=tf.float64)

with tf.GradientTape() as tape:
    tape.watch(x)
    y = target(x)

actual = tape.gradient(y, x)
expected = 4.4

print("forward:", y.numpy())
print("actual:", actual.numpy())
print("expected:", expected)

if abs(float(actual) - expected) > 1e-9:
    print("BUG REPRODUCED: tf.raw_ops.MatrixDiagV2 propagates gradients through ignored diagonal padding")
else:
    print("not reproduced")
```

### Relevant log output

```shell
forward: 1.5899999999999999
actual: 10.0
expected: 4.4
BUG REPRODUCED: tf.raw_ops.MatrixDiagV2 propagates gradients through ignored diagonal padding
```

### PR DESCRIPTION
**Pull request by twelfthlabor**
Closes #127235.

`MatrixDiagV2` always packs its diagonals left-aligned and has no align attr, but `_MatrixDiagV2Grad` extracted the gradient with `matrix_diag_part`'s default `RIGHT_LEFT` alignment, so gradients flowed through padded superdiagonal entries. The fix passes `align="LEFT_LEFT"` to the extraction.

Validation:
- Reported case: tape gradient 10.0 vs central finite difference 4.400000000126525 before; 4.4 == FD after.
- Regression test: fails before (failures=2, skipped=1, max_error 5.600000000000023) and passes after (OK, skipped=1), in graph and eager.
- Existing `diag_op_test`: same single pre-existing failure before and after.
- 22-case forward grid: V2 matches V3 LEFT_LEFT on every case.

The same alignment mismatch exists in the MatrixSetDiagV2 and MatrixDiagPartV2 grads; those fixes are verified separately and can land as follow-ups if maintainers want.

### CODE REVIEW

**Comment by keerthanakadiri** [comment]
Hi @twelfthlabor , Can you kindly sign the CLA? Thank you!

**Review by dmiltr3 — APPROVED** [review]
